src/main/java/org/jruby/ext/openssl/OCSPBasicResponse.java

```diff
@@ -0,0 +1,147 @@
+
+import java.security.SecureRandom;
+
+import org.bouncycastle.asn1.ocsp.BasicOCSPResponse;
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyFixnum;
+import org.jruby.RubyModule;
+import org.jruby.RubyObject;
+import org.jruby.RubyString;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.exceptions.RaiseException;
+import org.jruby.runtime.Arity;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.Visibility;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import static org.jruby.ext.openssl.OCSP._OCSP;
+
+/*
+ * An OpenSSL::OCSP::BasicResponse contains the status of a certificate
+ * check which is created from an OpenSSL::OCSP::Request. 
+ * A BasicResponse is more detailed than a Response.
+ * 
+ * @author lampad
+ */
+public class OCSPBasicResponse extends RubyObject {
+    private static final long serialVersionUID = 8755480816625884227L;
+    
+    private static ObjectAllocator BASICRESPONSE_ALLOCATOR = new ObjectAllocator() {
+        public IRubyObject allocate(Ruby runtime, RubyClass klass) {
+            return new OCSPBasicResponse(runtime, klass);
+        }
+    };
+    
+    public static void createBasicResponse(final Ruby runtime, final RubyModule _OCSP) {
+        RubyClass _BasicResponse = _OCSP.defineClassUnder("BasicResponse", runtime.getObject(), BASICRESPONSE_ALLOCATOR);
+        _BasicResponse.defineAnnotatedMethods(OCSPBasicResponse.class);
+    }
+    
+    private RubyString der;
+    private byte[] nonce;
+    private SecureRandom random = new SecureRandom();
+    private BasicOCSPResponse bcBasicOCSPResponse;
+
+    public OCSPBasicResponse(Ruby runtime, RubyClass metaClass) {
+        super(runtime, metaClass);
+    }
+    
+    @JRubyMethod(name = "initialize", visibility = Visibility.PRIVATE)
+    public IRubyObject initialize(final ThreadContext context, IRubyObject der) {        
+        if (der == null || der.isNil()) return this;
+        
+        this.der = StringHelper.readPossibleDERInput(context, der);
+        
+        return this;
+    }
+    
+    @JRubyMethod(name = "initialize", visibility = Visibility.PRIVATE)
+    public IRubyObject initialize() {
+        return this;
+    }
+    
+    @JRubyMethod(name = "add_nonce")
+    public OCSPBasicResponse add_nonce(final ThreadContext context, final IRubyObject[] args) {
+        Ruby runtime = context.getRuntime();
+        
+        if ( Arity.checkArgumentCount(runtime, args, 0, 1) == 0 ) {
+            nonce = generateNonce();
+        }
+        else if (args[0] == null || args[0].isNil()) {
+            nonce = generateNonce();
+        }
+        else {
+            RubyString input = (RubyString)args[0];
+            nonce = generateNonce(input.getBytes());
+        }
+        
+        return this;
+    }
+    
+    @JRubyMethod(name = "add_status")
+    public OCSPBasicResponse add_status(final ThreadContext context) {
+        //TODO: Implement
```

Comment:
Sorry, those were notes for myself to implement them before pushing. The final version should have everything implemented. I could rebase/squash my commits if that would help so people wouldn't even see this. Thoughts?